src/lib.rs

```diff
@@ -146,6 +150,16 @@ macro_rules! check_cpuid {
     } };
 }
 
+#[cfg(not(feature = "std"))]
+macro_rules! is_feature_supported {
```

Comment:
call this macro `is_x86_feature_detected`, then you no longer need the `#[cfg(feature = "std")] macro_rules! is_feature_supported` case